src/electionguard/discrete_log.py

```diff
@@ -58,29 +74,64 @@ async def discrete_log_async(
         if element in cache:
             return (cache[element], cache)
 
-        _cache = compute_discrete_log_cache(element, cache)
+        _cache = compute_discrete_log_cache(element, cache, max_exponent)
         return (_cache[element], _cache)
 
 
+def precompute_discrete_log_cache(
```

Comment:
There's a lot of new functionality here, should this have a covering unit test?

Reply:
Unit tests added